Bug: Live trade recorded in state even when execution fails

## Bug

When a live trade execution fails (e.g. due to the `float_to_wire` rounding error in #105), the bot still records the trade as successful in state. This causes:

1. **Phantom positions** — state shows positions that don't exist on-chain
2. **Inaccurate P&L** — portfolio value calculations are wrong
3. **Stale state** — subsequent cycles make decisions based on fake positions

## Reproduction

1. Trigger a live trade that fails during execution (e.g. size precision error)
2. Bot logs `[ERROR] Live execute returned error: ...`
3. Bot then logs `[INFO] BUY ETH: 0.466930 @ $2054.90` as if it succeeded
4. State records the position with `quantity: 0.466930`
5. No actual position exists on Hyperliquid

## Expected Behavior

If execution returns an error, the trade should be **rejected** — not recorded in state. The bot should:
- Log the failure clearly
- Skip updating cash/positions
- Optionally retry with corrected parameters

## Log

```
[INFO] Placing live buy ETH size=0.467123
[ERROR] Live execute returned error: ('float_to_wire causes rounding', 0.4671228074779094)
[INFO] BUY ETH: 0.466930 @ $2054.90 (fee $0.34, total $959.83)   ← should not happen
[INFO] Status: cash=$50.16 | positions=1 | trades=1               ← phantom position
```

## Related

- #105 (root cause of the execution failure)
- Both issues discovered during live Hyperliquid trading

--- a/scheduler/main.go
+++ b/scheduler/main.go
@@ -477,14 +477,19 @@ func main() {
 							if result, signalStr, price, ok := runOKXCheck(sc, prices, logger); ok {
 								prices[result.Symbol] = price
 								var execResult *OKXExecuteResult
+								liveExecFailed := false
 								if okxIsLive(sc.Args) && result.Signal != 0 {
 									if er, ok2 := runOKXExecuteOrder(sc, result, price, okxCash, okxPosQty, logger); ok2 {
 										execResult = er
+									} else {
+										liveExecFailed = true
 									}
 								}
-								mu.Lock()
-								trades, detail = executeOKXResult(sc, stratState, result, execResult, signalStr, price, logger)
-								mu.Unlock()
+								if !liveExecFailed {
+									mu.Lock()
+									trades, detail = executeOKXResult(sc, stratState, result, execResult, signalStr, price, logger)
+									mu.Unlock()
+								}
 							}
 						} else if sc.Platform == "robinhood" {
 							if result, signalStr, price, ok := runRobinhoodCheck(sc, prices, logger); ok {
@@ -532,27 +537,37 @@ func main() {
 						} else if result, signalStr, price, ok := runHyperliquidCheck(sc, prices, logger); ok {
 							prices[result.Symbol] = price
 							var execResult *HyperliquidExecuteResult
+							liveExecFailed := false
 							if hyperliquidIsLive(sc.Args) && result.Signal != 0 {
 								if er, ok2 := runHyperliquidExecuteOrder(sc, result, price, hlCash, hlPosQty, logger); ok2 {
 									execResult = er
+								} else {
+									liveExecFailed = true
 								}
 							}
-							mu.Lock()
-							trades, detail = executeHyperliquidResult(sc, stratState, result, execResult, signalStr, price, logger)
-							mu.Unlock()
+							if !liveExecFailed {
+								mu.Lock()
+								trades, detail = executeHyperliquidResult(sc, stratState, result, execResult, signalStr, price, logger)
+								mu.Unlock()
+							}
 						}
 					case "futures":
 						if result, signalStr, price, ok := runTopStepCheck(sc, prices, logger); ok {
 							prices[result.Symbol] = price
 							var execResult *TopStepExecuteResult
+							liveExecFailed := false
 							if topstepIsLive(sc.Args) && result.Signal != 0 {
 								if er, ok2 := runTopStepExecuteOrder(sc, result, price, tsCash, tsContracts, logger); ok2 {
 									execResult = er
+								} else {
+									liveExecFailed = true
 								}
 							}
-							mu.Lock()
-							trades, detail = executeTopStepResult(sc, stratState, result, execResult, signalStr, price, logger)
-							mu.Unlock()
+							if !liveExecFailed {
+								mu.Lock()
+								trades, detail = executeTopStepResult(sc, stratState, result, execResult, signalStr, price, logger)
+								mu.Unlock()
+							}
 						}
 					default:
 						logger.Error("Unknown strategy type: %s", sc.Type)

